actionpack/lib/action_controller/metal/implicit_render.rb

```diff
@@ -2,14 +2,26 @@ module ActionController
   module ImplicitRender
     def send_action(method, *args)
       ret = super
-      default_render unless performed?
-      ret
+      return ret if performed?
+
+      lookup_context.variants = request.variant
+
+      if template_exists?(method, _prefixes)
+        default_render
+      else
+        default_head
+      end
     end
 
     def default_render(*args)
       render(*args)
     end
 
+    def default_head(*)
```

Comment:
I made a comment about that on the original comment you did. I think we should not change method_for_action because it isn't in the use case and could very easily lead to silent confusion in certain cases. I'll pull it back into the `send_action` method.